core/src/main/scala/gnieh/diffson/DynamicProgLcs.scala

```diff
@@ -28,16 +28,25 @@ class DynamicProgLcs[T] extends Lcs[T] {
     } else {
       // we try to reduce the problem by stripping common suffix and prefix
       val (prefix, middle1, middle2, suffix) = splitPrefixSuffix(seq1, seq2, low1, low2)
+      val indexedMiddle1 : IndexedSeq[T] = middle1.toIndexedSeq
+      val indexedMiddle2 : IndexedSeq[T] = middle2.toIndexedSeq
       val offset = prefix.size
-      val lengths = Array.ofDim[Int](middle1.size + 1, middle2.size + 1)
+      val lengths: Array[Array[Int]] = Array.ofDim[Int](middle1.size + 1, middle2.size + 1)
       // fill up the length matrix
-      for {
-        i <- 0 until middle1.size
-        j <- 0 until middle2.size
-      } if (middle1(i) == middle2(j))
-        lengths(i + 1)(j + 1) = lengths(i)(j) + 1
-      else
-        lengths(i + 1)(j + 1) = math.max(lengths(i + 1)(j), lengths(i)(j + 1))
+      locally {
+        var i: Int = 0
+        while(i < indexedMiddle1.size) {
+          var j: Int = 0
+          while(j < indexedMiddle2.size) {
+            if (indexedMiddle1(i).equals(indexedMiddle2(j)))
+              lengths(i + 1)(j + 1) = lengths(i)(j) + 1
+            else
+              lengths(i + 1)(j + 1) = math.max(lengths(i + 1)(j), lengths(i)(j + 1))
+            j += 1
+          }
+          i += 1
+        }
+      }
```

Comment:
Does the use of while loops really improve the performance (all other changes being done)? If yes, I generally prefer tail-recursive implementation of while loops. Can you please change it to its recursive equivalent?